wolfcrypt/src/pkcs7.c

```diff
@@ -1668,6 +1673,28 @@ int wc_PKCS7_VerifySignedData(PKCS7* pkcs7, byte* pkiMsg, word32 pkiMsgSz)
     if (GetSequence(pkiMsg, &idx, &length, pkiMsgSz) < 0)
         return ASN_PARSE_E;
 
+    if (length == 0 && pkiMsg[idx-1] == 0x80) {
+#ifdef ASN_BER_TO_DER
+        int len;
+
+        len = wc_BerToDer(pkiMsg, pkiMsgSz, NULL);
+        if (len < 0)
+            return len;
+        pkcs7->der = XMALLOC(len, pkcs7->heap, DYNAMIC_TYPE_PKCS7);
+        if (pkcs7->der == NULL)
+            return MEMORY_E;
+        wc_BerToDer(pkiMsg, pkiMsgSz, pkcs7->der);
```

Comment:
We should check the return code here, there's no guarantee in the future that if the first call succeeds on the length check that the second call will succeed as well.  Also a good static analyzer should warn on this.